File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        // BRUTE FORCE

        // int maxWater = 0;
        // for (int i=0; i<heights.size(); i++){
        //     for (int j=i+1; j<heights.size(); j++){
        //         int water = min(heights[i], heights[j]) * (j-i);
        //         maxWater = max(maxWater, water);
        //     }
        // }
        // return maxWater;

        // TWO POINTER

        int i=0;
        int j=heights.size() - 1;
        int maxWater = 0;
        while(i < j){
            int water = min(heights[i], heights[j]) * (j-i);
            maxWater = max(maxWater, water);
            if (heights[i]<heights[j]) i++;
            else j--;
        }
        return maxWater;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        """
        BRUTE FORCE
        """
        n = len(prices)
        maxProfit = 0
        for i in range(n):
            for j in range(i+1, n):
                profit = prices[j] - prices[i]
                maxProfit = max(profit, maxProfit)
        return maxProfit